crates/bevy_render/src/texture/texture_attachment.rs

```diff
@@ -0,0 +1,124 @@
+            is_first_call: Arc::new(AtomicBool::new(true)),
+        }
+    }
+
+    /// Get this texture view as an attachment. The attachment will be cleared with a value of
+    /// `clear_color` if this is the first time calling this function, otherwise it will be loaded.
+    ///
+    /// The returned attachment will always have writing enabled (`store: StopOp::Load`).
+    pub fn get_attachment(&self) -> RenderPassColorAttachment {
+        if let Some(resolve_target) = self.resolve_target.as_ref() {
+            let first_call = self.is_first_call.fetch_and(false, Ordering::SeqCst);
+
+            RenderPassColorAttachment {
+                view: &resolve_target.default_view,
+                resolve_target: Some(&self.texture.default_view),
+                ops: Operations {
+                    load: if first_call {
+                        LoadOp::Clear(self.clear_color.into())
+                    } else {
+                        LoadOp::Load
+                    },
+                    store: StoreOp::Store,
+                },
+            }
+        } else {
+            self.get_unsampled_attachment()
+        }
+    }
+
+    /// Get this texture view as an attachment, without the resolve target. The attachment will be cleared with
+    /// a value of `clear_color` if this is the first time calling this function, otherwise it will be loaded.
+    ///
+    /// The returned attachment will always have writing enabled (`store: StopOp::Load`).
+    pub fn get_unsampled_attachment(&self) -> RenderPassColorAttachment {
+        let first_call = self.is_first_call.fetch_and(false, Ordering::SeqCst);
+
+        RenderPassColorAttachment {
+            view: &self.texture.default_view,
+            resolve_target: None,
+            ops: Operations {
+                load: if first_call {
+                    LoadOp::Clear(self.clear_color.into())
+                } else {
+                    LoadOp::Load
+                },
+                store: StoreOp::Store,
+            },
+        }
+    }
+
+    pub(crate) fn mark_as_cleared(&self) {
+        self.is_first_call.store(false, Ordering::SeqCst);
+    }
+}
+
+/// A wrapper for a [TextureView] that is used as a depth-only [RenderPassDepthStencilAttachment].
+pub struct DepthAttachment {
+    pub view: TextureView,
+    clear_value: f32,
+    is_first_call: Arc<AtomicBool>,
+}
+
+impl DepthAttachment {
+    pub fn new(view: TextureView, clear_value: Option<f32>) -> Self {
+        Self {
+            view,
+            clear_value: clear_value.unwrap(),
+            is_first_call: Arc::new(AtomicBool::new(clear_value.is_some())),
+        }
+    }
+
+    /// Get this texture view as an attachment. The attachment will be cleared with a value of
+    /// `clear_value` if this is the first time calling this function, and a clear value was provided,
+    /// otherwise it will be loaded.
+    pub fn get_attachment(&self, store: bool) -> RenderPassDepthStencilAttachment {
```

Comment:
Was this done to avoid a dependency on wgpu in other crates? If that's the case then maybe we could wrap it in our own enum? I'd like to not have bare boolean arguments if at all possible.

Reply:
Leftover API from before wgpu changed this, fixed.